core/types/access_list_tx.go

```diff
@@ -97,7 +97,7 @@ func newAccessListTransaction(chainId *big.Int, nonce uint64, to *common.Address
 }
 
 func (tx *AccessListTransaction) ChainId() *big.Int       { return tx.Chain }
-func (tx *AccessListTransaction) Protected() bool         { return true }
+func (tx *AccessListTransaction) Protected() bool         { return tx.Chain.Cmp(common.Big0) == 0 }
```

Comment:
oldie but goldie: you can use `tx.Chain.BitLen() == 0` instead

Reply:
But shouldn't it be the other way around (both in the current code and my example?) `!= 0` ?

Reply:
Ah, good catch. I had fixed this locally, but managed to not save the changes before committing. I'll update to `tx.Chain.BitLen() != 0` instead of using `Cmp`.